tools/terraform/fleetdm_client/fleetdm_client.go

```diff
@@ -0,0 +1,276 @@
+package fleetdm_client
+
+// This file gives us a nice API to use to call FleetDM's API. It's focused
+// only on teams.
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var prefix = "/api/v1/fleet"
+var teamPrefix = prefix + "/teams"
+
+type Team struct {
+	Name         string      `json:"name"`
+	ID           int64       `json:"id"`
+	Description  string      `json:"description"`
+	AgentOptions interface{} `json:"agent_options"`
+	Scripts      interface{} `json:"scripts"`
+	Secrets      []struct {
+		Secret  string    `json:"secret"`
+		Created time.Time `json:"created_at"`
+		TeamID  int       `json:"team_id"`
+	} `json:"secrets"`
+}
+
+type TeamCreate struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+}
+
+type TeamGetResponse struct {
+	Team struct {
+		Team
+	} `json:"team"`
+}
+
+type TeamQueryResponse struct {
+	Teams []Team `json:"teams"`
+}
+
+// FleetDMClient is a FleetDM HTTP client that overrides http.DefaultClient.
+type FleetDMClient struct {
+	*http.Client
+	URL    string
+	APIKey string
+}
+
+// NewFleetDMClient creates a new instance of FleetDMClient with the provided
+// URL and API key.
+func NewFleetDMClient(url, apiKey string) *FleetDMClient {
+	return &FleetDMClient{
+		Client: http.DefaultClient,
+		URL:    url,
+		APIKey: apiKey,
+	}
+}
+
+// Do will add necessary headers and call the http.Client.Do method.
+func (c *FleetDMClient) do(req *http.Request, query string) (*http.Response, error) {
+	// Add the API key to the request header
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", c.APIKey))
+	req.Header.Add("Accept", `application/json`)
+	// Set the request URL based on the client URL
+	req.URL, _ = url.Parse(c.URL + req.URL.Path)
```

Comment:
This concatenation of URL and path could result in issues with edge cases depending on what is provided in `c.URL`. I think it would be best to use:
```
        baseURL, err := url.Parse(c.URL)
        if err != nil { ... handle error ... }
	req.URL = baseURL.JoinPath(req.URL.Path)
```

Reply:
Yep that's cleaner.

Reply:
Done.